code/go/0chain.net/blobbercore/handler/handler.go

```diff
@@ -330,15 +330,21 @@ func UpdateAttributesHandler(ctx context.Context, r *http.Request) (interface{},
 	return response, nil
 }
 
-func CalculateHashHandler(ctx context.Context, r *http.Request) (interface{}, error) {
-	ctx = setupHandlerContext(ctx, r)
+func CalculateHashHandler(svc *blobberGRPCService) func(ctx context.Context, r *http.Request) (interface{}, error) {
+	return func(ctx context.Context, r *http.Request) (interface{}, error) {
+		ctx = setupHandlerGRPCContext(ctx, r)
 
-	response, err := storageHandler.CalculateHash(ctx, r)
-	if err != nil {
-		return nil, err
-	}
+		response, err := svc.CalculateHash(ctx, &blobbergrpc.CalculateHashRequest{
+			Allocation: mux.Vars(r)["allocation"],
+			Paths:      r.FormValue("paths"),
+			Path:       r.FormValue("path"),
+		})
+		if err != nil {
+			return nil, err
+		}
 
-	return response, nil
+		return response, nil
```

Comment:
The response needs to be converted to how it is in the original or it will break existing clients